com.unity.render-pipelines.universal/Runtime/RenderingUtils.cs

```diff
@@ -142,30 +141,65 @@ public static void SetViewAndProjectionMatrices(CommandBuffer cmd, Matrix4x4 vie
-                CoreUtils.SetRenderTarget(cmd, destination, colorLoadAction, colorStoreAction, ClearFlag.None, Color.clear);
-                cmd.DrawProcedural(Matrix4x4.identity, material, passIndex, MeshTopology.Quads, 4, 1, null);
-            }
-            else
-            {
-                CoreUtils.SetRenderTarget(cmd, destination, colorLoadAction, colorStoreAction, ClearFlag.None, Color.clear);
-                cmd.Blit(source.nameID, BuiltinRenderTextureType.CurrentActive, material, passIndex);
-            }
+            Vector2 viewportScale = source.useScaling ? new Vector2(source.rtHandleProperties.rtHandleScale.x, source.rtHandleProperties.rtHandleScale.y) : Vector2.one;
+            CoreUtils.SetRenderTarget(cmd, destination, loadAction, storeAction, ClearFlag.None, Color.clear);
+            cmd.SetViewport(viewport);
+            Blitter.BlitTexture(cmd, source, viewportScale, material, passIndex);
+        }
+
+        internal static void Blit(CommandBuffer cmd,
+            RTHandle source,
+            Rect viewport,
+            RTHandle destinationColor,
+            RenderBufferLoadAction colorLoadAction,
+            RenderBufferStoreAction colorStoreAction,
+            RTHandle destinationDepthStencil,
+            RenderBufferLoadAction depthStencilLoadAction,
+            RenderBufferStoreAction depthStencilStoreAction,
+            ClearFlag clearFlag,
+            Color clearColor,
+            Material material,
+            int passIndex = 0)
+        {
+            Vector2 viewportScale = source.useScaling ? new Vector2(source.rtHandleProperties.rtHandleScale.x, source.rtHandleProperties.rtHandleScale.y) : Vector2.one;
+            CoreUtils.SetRenderTarget(cmd,
+                destinationColor, colorLoadAction, colorStoreAction,
+                destinationDepthStencil, depthStencilLoadAction, depthStencilStoreAction,
+                clearFlag, clearColor); // implicit depth=1.0f stencil=0x0
+            cmd.SetViewport(viewport);
+            Blitter.BlitTexture(cmd, source, viewportScale, material, passIndex);
+        }
+
+        internal static void FinalBlit(
+            CommandBuffer cmd,
+            CameraData cameraData,
+            bool isRenderToBackBufferTarget,
+            RTHandle source,
+            RTHandle destination,
+            RenderBufferLoadAction loadAction,
+            RenderBufferStoreAction storeAction,
+            Material material, int passIndex)
+        {
+            Vector2 viewportScale = source.useScaling ? new Vector2(source.rtHandleProperties.rtHandleScale.x, source.rtHandleProperties.rtHandleScale.y) : Vector2.one;
+
+            // We y-flip if
+            // 1) we are blitting from render texture to back buffer(UV starts at bottom) and
+            // 2) renderTexture starts UV at top
+            bool yflip = isRenderToBackBufferTarget && cameraData.targetTexture == null && SystemInfo.graphicsUVStartsAtTop;
```

Comment:
Is it possible to figure out `isRenderToBackBufferTarget` here?

Reply:
I was not able to do it. The logic ends up being a bit different in each case.

* Final Blit Pass: `bool isRenderToBackBufferTarget = !cameraData.isSceneViewCamera || cameraData.xr.enabled;`
* SSAO: `bool isRenderToBackBufferTarget = !cameraData.isSceneViewCamera || cameraData.xr.enabled;`
* Post Process Pass
    * Render: `bool isRenderToBackBufferTarget = cameraData.xr.enabled ? (cameraTargetHandle == cameraData.xr.renderTarget) : (m_UseSwapBuffer ? m_ResolveToScreen : m_Destination.nameID == BuiltinRenderTextureType.CameraTarget);`
    * RenderFinalPass: `bool isRenderToBackBufferTarget = cameraData.xr.enabled ? (m_CameraTargetHandle == cameraData.xr.renderTarget) : true;`

Reply:
The main issue I see to merging the logic is the Render in PostProcess pass which has a branch with an extra blit `if (!m_ResolveToScreen && !m_UseSwapBuffer)`

>                // TODO: Implement swapbuffer in 2DRenderer so we can remove this
>                // For now, when render post-processing in the middle of the camera stack (not resolving to screen)
>                // we do an extra blit to ping pong results back to color texture. In future we should allow a Swap of the current active color texture
>                // in the pipeline to avoid this extra blit.